src/Bridge/Doctrine/MongoDbOdm/Filter/SearchFilter.php

```diff
@@ -0,0 +1,207 @@
+                $this->logger->notice('Invalid filter ignored', [
+                    'exception' => new InvalidArgumentException(sprintf('Values for field "%s" are not valid according to the doctrine type.', $field)),
+                ]);
+
+                return;
+            }
+
+            $strategy = $this->properties[$property] ?? self::STRATEGY_EXACT;
+
+            // prefixing the strategy with i makes it case insensitive
+            if (0 === strpos($strategy, 'i')) {
+                $strategy = substr($strategy, 1);
+                $caseSensitive = false;
+            }
+
+            if (1 === \count($values)) {
+                $aggregationBuilder
+                    ->match()
+                    ->field($matchField)
+                    ->equals($this->addEqualityMatchStrategy($strategy, $field, $values[0], $caseSensitive, $metadata));
+
+                return;
+            }
+
+            $inValues = [];
+            foreach ($values as $inValue) {
+                $inValues[] = $this->addEqualityMatchStrategy($strategy, $field, $inValue, $caseSensitive, $metadata);
+            }
+            $aggregationBuilder
+                ->match()
+                ->field($matchField)
+                ->in($inValues);
+        }
+
+        // metadata doesn't have the field, nor an association on the field
+        if (!$metadata->hasAssociation($field)) {
+            return;
+        }
+
+        $values = array_map([$this, 'getIdFromValue'], $values);
+
+        if (!$this->hasValidValues($values, $this->getDoctrineFieldType($property, $resourceClass))) {
+            $this->logger->notice('Invalid filter ignored', [
+                'exception' => new InvalidArgumentException(sprintf('Values for field "%s" are not valid according to the doctrine type.', $property)),
+            ]);
+
+            return;
+        }
+
+        if (1 === \count($values)) {
+            $aggregationBuilder
+                ->match()
+                ->field("$matchField.\$id")
+                ->equals($values[0]);
+        } else {
+            $aggregationBuilder
+                ->match()
+                ->field("$matchField.\$id")
+                ->in($values);
+        }
+    }
+
+    /**
+     * Add equality match stage according to the strategy.
+     *
+     * @throws InvalidArgumentException If strategy does not exist
+     *
+     * @return Regex|string
+     */
+    private function addEqualityMatchStrategy(string $strategy, string $field, $value, bool $caseSensitive, ClassMetadata $metadata)
+    {
+        $type = $metadata->getTypeOfField($field);
+
+        switch ($strategy) {
+            case MongoDbType::STRING !== $type:
+                return MongoDbType::getType($type)->convertToDatabaseValue($value);
+            case null:
+            case self::STRATEGY_EXACT:
+                return $caseSensitive ? $value : new Regex("^$value$", $caseSensitive ? '' : 'i');
```

Comment:
The second ternary within the `Regex` constructor is useless since it already must have evaluated to `false` in order to hit the constructor.